pyseq.py

```diff
@@ -128,6 +139,24 @@ def __init__(self, item):
         self.head = self.name
         self.tail = ''
 
+    def __eq__(self, other):
```

Comment:
Hmm, I think you're right. I commented out the `__eq__` override and could not reproduce a negative result

```python
>>> i1 = pyseq.Item("foo.bar")
>>> i2 = pyseq.Item("foo.bar")
>>> i1 == i2
True
```
And unit tests pass, so maybe it's not needed after-all.